### DISCUSSION TITLE
About the output values?

### DISCUSSION THREAD

**Discussion by OnlyBelter** [brianhie/scanorama#85] [General]
Hi, very impressive tool.

What's the output should be expected?

If I input log-transformed data, does it output log-transformed data? I mean if I want to use it for downstream analysis, such as differential expression analysis, then how can I calculate fold change (log or non-log)?

What I have seen is that the output values are much smaller than what I input. How can I recover the corrected values back as much as like input data (such as counts per million, CPM)?

**Comment by brianhie** [comment]
Hi @OnlyBelter, thanks for reaching out! What I'd recommend is to use the low dimensional Scanorama embeddings (from `scanorama.integrate()` for the main global analysis, e.g., visualizing with UMAP, constructing trajectories, and clustering. To be super conservative, for looking at differential gene expression, you can use the original gene expression values with an optional batch correction step.

See this comment and the surrounding discussion for more clarity: https://github.com/theislab/anndata/issues/458#issuecomment-730709926. Basically, the modified values from `scanorama.correct()` are in an arbitrary space that just tries to make more similar cell types closer to each other, but this might also violate some of the assumptions of differential expression analysis. There are tools like [Combat](https://scanpy.readthedocs.io/en/stable/api/scanpy.pp.combat.html) that can produce batch effect corrected matrices using a more conservative strategy and that appears to benchmark well with simulated data (see https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1850-9).

Hope that helps!

**Comment by wconnell** [comment]
This is interesting and a useful discussion.

I'm experiencing failure using corrected expression features for supervised learning. The same model converges/generalizes when trained and evaluated on uncorrected, preprocessed data. Is there evidence that corrected features will produce comparable results for supervised learning? Is my assumption wrong that corrected gene expression values reflect the distributional characteristics of uncorrected features? 

I understand the embeddings may be more useful for this task but I am specifically wondering about the utility of "corrected features."

**Comment by brianhie** [comment]
@wconnell I think the consensus in the field now is that corrected features and the integrated embeddings are useful when computing distances among cells, e.g., for constructing a KNN graph for clustering, visualization, etc. When interpreting the actual feature values, e.g., for differential expression analysis, it's best to be conservative. I'm not sure about your specific supervised task and how you are defining "failure" etc., but I'd also recommend a conservative approach to dealing with potential batch effect.
